cuda_bindings/cuda/bindings/_bindings/cydriver.pyx.in

```diff
@@ -490,6 +490,7 @@ cdef int cuPythonInit() except -1 nogil:
     global __cuPythonInit
     if __cuPythonInit:
         return 0
+    __cuPythonInit = True
```

Comment:
Without moving this back here it deadlocks in an infinite loop related to using `cuGetProcAddress` for resolving functions and `cuGetProcAddress` isn't defined in this file where I'm not even clear how it's resolved in here

Reply:
I am not sure what could cause a deadlock in the CI? We don't have multithreading tests 👀

This change smells because multiple threads can overwrite the bool.

Reply:
following up in this thread: https://github.com/NVIDIA/cuda-python/pull/835#discussion_r2279149765